aiida/cmdline/commands/cmd_calcjob.py

```diff
@@ -267,8 +279,9 @@ def calcjob_cleanworkdir(calcjobs, past_days, older_than, computers, force):
         transport = orm.AuthInfo.objects.get(dbcomputer_id=computer.id, aiidauser_id=user.id).get_transport()
 
         with transport:
-            for path in paths:
+            for (calcjob_id, path) in paths:
                 clean_remote(transport, path)
                 counter += 1
+                load_node(calcjob_id).set_extra('workdir_cleaned', True)
```

Comment:
If we are going to make part of the way things work, we should really make this logic part of the `RemoteData` class, i.e., the `_clean` method should be responsible for setting this extra. Now the `clean_remote` function doesn't call this method because it doesn't allow specifying the transport making it very inefficient. But I would simply update `_clean` to be
```
def _clean(self, transport=None)
```
to allow passing a transport. This way you can use this method in the CLI, allowing to centralize the setting of the `cleaned` extra, and it is backwards compatible.

One problem we would have to keep in mind is really that we should maybe check in `_clean` that the passed `transport` is configured for the `Computer` that is the same as the computer of the `RemoteData`.

Reply:
good point. this also means we the `cleaned` should be stored in `extras` of the `RemoteData` not `CalcJobNode`. So I have updated the code to that instead.